index.d.ts

```diff
@@ -110,49 +169,46 @@ export interface BlockOrderOptions {
   order?: BlockOrder
 }
 
-export interface IDagula {
-  /**
-   * Get a complete DAG by root CID.
-   */
+/** @deprecated Use `BlockService`, `DagService` and `UnixfsService` interface instead. */
+export interface IDagula extends BlockService, DagService, UnixfsService {}
+
+export interface BlockService {
+  /** Get a single block. */
+  getBlock (cid: UnknownLink|string, options?: AbortOptions): Promise<Block>
+  /** Retrieve information about a block. */
+  statBlock (cid: UnknownLink|string, options?: AbortOptions): Promise<BlockStat>
+  /** Stream bytes from a single block. */
+  streamBlock (cid: UnknownLink|string, options?: AbortOptions & RangeOptions): Promise<ReadableStream<Uint8Array>>
+}
+
+export interface DagService {
+  /** Get a complete DAG by root CID. */
   get (cid: UnknownLink|string, options?: AbortOptions & BlockOrderOptions): AsyncIterableIterator<Block>
-  /**
-   * Get a DAG for a cid+path.
-   */
+  /** Get a DAG for a cid+path. */
   getPath (cidPath: string, options?: AbortOptions & DagScopeOptions & EntityBytesOptions & BlockOrderOptions): AsyncIterableIterator<Block>
-  /**
-   * Get a single block.
-   */
-  getBlock (cid: UnknownLink|string, options?: AbortOptions): Promise<Block>
-  /**
-   * Get UnixFS files and directories.
-   */
+}
+
+export interface UnixfsService {
+  /** Get UnixFS files and directories. */
   getUnixfs (path: UnknownLink|string, options?: AbortOptions): Promise<UnixFSEntry>
-  /**
-   * Emit nodes for all path segements and get UnixFS files and directories.
-   */
+  /** Emit nodes for all path segements and get UnixFS files and directories. */
   walkUnixfsPath (path: UnknownLink|string, options?: AbortOptions): AsyncIterableIterator<UnixFSEntry>
 }
 
-export declare class Dagula implements IDagula {
+export declare class Dagula implements BlockService, DagService, UnixfsService {
```

Comment:
Note: I'm planning on separating this into multiple classes that implement the respective interfaces. However it's going to take a bit of time because the unixfs specific stuff is all mixed up with the generic DAG traversal stuff.